Ui development

justakit opened the pull request.

justakit committed:
Merge branch 'master' into UI_Development

justakit committed:
Merge branch 'master' into UI_Development

justakit committed:
Change message types according to wiki update

Reworked commands for unsuccessfully responses

Fixed a lot of qml-warnings

Fixed uninitialized CurrentUsername variable

Added request for contact's list

Improved contact's list view

bohdanlisovskyi commented:
fix conflict

8tomat8 commented on client/ui/channelsresolver.go:
```diff
@@ -15,25 +15,32 @@ func channelsResolver() {
 	for {
 		select {
 		case <-listener.QuitChannel:
+			//loger.Log.Info("Quit channel")
```
Why are you added commented logging functions? Uncomment them or remove

8tomat8 commented on client/ui/channelsresolver.go:
```diff
@@ -15,25 +15,32 @@ func channelsResolver() {
 			case config.GetConfig().MessageCommand.LoginIsSucc:
 				loginIsSuccessfully(msg)
-			case config.GetConfig().MessageCommand.LoginIsNotSucc:
+			default:
```
Here switch is excess. It would be more efficient (but less readable) to use `if`

8tomat8 commented on client/ui/contacts.go:
```diff
@@ -102,3 +109,37 @@ func contactsData(index *core.QModelIndex, role int) *core.QVariant {
+	if err != nil {
+		loger.Log.Warningf("Can`t marshal contacts message body. %s", err)
+		qmlStatus.SendStatus("Can not load contacts.")
+		return
```
It would be useful to return error. Imagine if error happens here,waht will user do without contacts?

8tomat8 commented on client/ui/contacts.go:
```diff
@@ -102,3 +109,37 @@ func contactsData(index *core.QModelIndex, role int) *core.QVariant {
+	}
+	if err := connection.WriteMessage(websocket.TextMessage, marshaledMessage); err != nil {
+		loger.Log.Warningf("Can not send contacts request. %s", err)
+		qmlStatus.SendStatus("Can not load contacts.")
```
This message occurs here for third time. Move it to constants

8tomat8 reviewed (approved):
PR looks pretty good. Resolve merge conflicts. And please pay attention to comments

justakit committed:
Added constants

Remove comments

justakit committed:
Added timer for waiting response

Change switch to if statements in channelsresolver where it was needed

Added busy indicator

Fix bug with highlighting current contact
